bootstrap/parser.go

```diff
@@ -242,15 +243,70 @@ type seqParser struct {
 	put     putter
 }
 
+func identFromTerm(term Term) string {
+	switch x := term.(type) {
+	case Named:
+		if x.Name != "" {
+			return x.Name
+		}
+		return identFromTerm(x.Term)
+	case Rule:
+		return string(x)
+	case Quant:
+		return identFromTerm(x.Term)
+	}
+	return ""
+}
+
+func nodesEqual(a, b interface{}) bool {
+	aType := reflect.TypeOf(a)
+	bType := reflect.TypeOf(b)
+	if aType == bType {
+		switch a := a.(type) {
+		case parser.Node:
+			b := b.(parser.Node)
+			diff := parser.NewNodeDiff(&a, &b)
+			if diff.Equal() {
+				return true
+			}
+		case parser.Scanner:
+			b := b.(parser.Scanner)
+			if a.String() == b.String() {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 func (p *seqParser) Parse(input *parser.Scanner, output interface{}) (out error) {
 	defer enterf("%s: %T %[2]v", p.rule, p.t).exitf("%v %v", &out, output)
 	result := make([]interface{}, 0, len(p.parsers))
 	furthest := *input
-	for _, parser := range p.parsers {
+
+	for _, item := range p.parsers {
 		var v interface{}
-		if err := parser.Parse(input, &v); err != nil {
-			*input = furthest
-			return err
+		if ref, ok := item.(*REF); ok {
+			for i := 0; i < len(result); i++ {
```

Comment:
```
for i, v := range result {
```

Reply:
i think i intended to iterate through the results slice backwards, not sure if thats needed, switching.